amm/contracts/factory/Cargo.toml

```diff
@@ -7,10 +7,10 @@ edition = "2021"
 [dependencies]
 ink = { version = "=4.3.0", default-features = false }
 
-scale = { package = "parity-scale-codec", version = "3", default-features = false, features = [
+scale = { package = "parity-scale-codec", version = "=3.6.9", default-features = false, features = [
     "derive",
 ] }
-scale-info = { version = "2.9", default-features = false, features = [
+scale-info = { version = "=2.10.0", default-features = false, features = [
```

Comment:
Why is fixing all those versions needed?

Reply:
The rust version for this project is fixed to `1.72.1`. `scale-info` has `derive_more` as a dependency, which in `scale-info` > `2.11.3` requires rust `1.75.0` or newer.